mesonbuild/interpreter.py

```diff
@@ -3739,6 +3740,15 @@ def func_add_test_setup(self, node, args, kwargs):
                                                              timeout_multiplier=timeout_multiplier,
                                                              env=env)
 
+    @noKwargs
+    @stringArgs
+    def func_set_default_test_setup(self, node, args, kwargs):
+        if len(args) != 1:
+            raise InterpreterException('Set_default_test_setup needs one argument for the setup name.')
+        if self.build.test_setup_default_name is not None:
+            raise InterpreterException('Set_default_test_setup can only be called once.')
+        self.build.test_setup_default_name = args[0]
```

Comment:
This should verify that `args[0]` is a test setup that already has been defined and if not, fail with an error.